zdm/iteration.py

```diff
@@ -965,39 +996,55 @@ def cube_likelihoods(grids,surveys,psetmins,psetmaxes,npoints,
             # provided that current is updated properly this does not care about re-ordering
             for j,n in enumerate(current):
                 if active[j]==0 or i==0: #only update if we are not minimising or it's the first time
-                    pset[j]=psetvals[j][n]
-            
+                    vparams[vp_keys[j]] = vparam_dict[vp_keys[j]]['vals'][n]
+                    #pset[j]=psetvals[j][n]
+
+            # Update the state
+            #state.update_params(vparams)
+            print(f"vparams: {vparams}")
+
             ### minimise if appropriate ### 
             if minimise:
-                C_ll,C_p=my_minimise(pset,grids,surveys,disable=disable,psnr=True,PenParams=None,Verbose=False)
-                toset=np.where(active==1)
-                for s in toset:
-                    pset[s]=C_p[s]
+                #C_ll,C_p=my_minimise(pset,grids,surveys,
+                C_ll = my_minimise(vparams,grids,surveys,
```

Comment:
one might consider that the minimise functionality is deprecated. But let's keep it here for now.